### ISSUE TITLE
update sinon usages

### ISSUE THREAD

**Issue by boneskull** [mochajs/mocha#4303]
As of v5.0.0 (we are on 9.x, so it's been there awhile), [sinon](https://npm.im/sinon) is a "sandbox", which means `sinon.createSandbox()` is no longer needed.

In our unit tests, we have a lot of this:

```js
let sandbox;
beforeEach(function() {
  sandbox = sinon.createSandbox();
});
afterEach(function() {
  sandbox.restore();
});
```

We can change this to:

```js
afterEach(function() {
  sinon.restore();
});
```

...and replace all instances of `sandbox` (may sometimes be called `sbx`?) with `sinon`.

See [Sinon migration guide](https://sinonjs.org/guides/migrating-to-5.0).

**Comment by sharath2106** [comment]
I'd like to work on this if this is not taken.

**Comment by nicojs** [comment]
@sharath2106 go right ahead!  👍

**Comment by juergba** [comment]
closed by #4312
